### PR TITLE
Proposal: Add rich data to the channel communications

### PR DESCRIPTION
**Pull request by bunnymatic**
* Add extra param `data` to send_to_player method
* handle it in clients
* add `/edit` method for `Behavoirs.God`
* update frontends to handle the new structure (React and Slack)
* React frontend still does not respond to edit but it will at least handle the new structured response

### CODE REVIEW

File: lib/doro/comms.ex [hunk 1]
```diff
@@ -3,7 +3,7 @@ defmodule Doro.Comms do
   Functions for sending output to players.
   """
   def send_to_player(player = %Doro.Entity{}, s) do
-    Phoenix.PubSub.broadcast(Doro.PubSub, "player-session:#{player.id}", {:send, s})
+    send_output_to_player(player, s)
```
Comment (ingar) [inline lib/doro/comms.ex]: Would it be simpler to augment the public methods in this module to take an optional `data` parameter?  That way, the "textual" representation would always exist, and you could optionally add structured data. Something like:

```elixir
def send_to_player(player, s, data \\ %{}) do
  payload = %{text: s, data: data}  # or Map.put(data, :text, s)
  broadcast(Doro.PubSub, "player-session:#{player.id}", {:send, payload})
end
```

So a command "/edit sewing machine" would do the correct thing on all clients:
```elixir
send_to_player(p, "You concentrate on #{definite(entity)}, until you feel silly.", %{type: :edit, entity: marshal(entity)})
```

With this, you'd have to also modify the [slack session](https://github.com/dorodango/doro/blob/8b136d7722c9c5523ba43ec54d3a7cd27357c4a1/lib/doro/transports/slack/session.ex#L29-L30) and the [web session](https://github.com/dorodango/doro/blob/8b136d7722c9c5523ba43ec54d3a7cd27357c4a1/lib/doro/session.ex#L41-L42) to use the new payload.

Comment (bunnymatic) [reply lib/doro/comms.ex]: yep.  sounds good.  as we discussed in person, better to have a response for UI's that don't support the action.

File: lib/doro/behaviors/god.ex [hunk 2]
```diff
@@ -20,4 +21,19 @@ defmodule Doro.Behaviors.God do
...
+    |> Enum.at(0)
+    rest_present = (rest != nil) && (String.length(rest |> String.trim()) > 0)
+    rest_present && found_entity
+  end
+
+  interact("/edit", %{player: player, rest: name}) do
+    entity = Doro.World.get_entities([in_location(player[:location]), named(name)]) |> Enum.at(0)
+    send_to_player(player,
+      "You concentrate on #{definite(entity)}.  Nothing happens and you feel silly.",
+      entity
```
Comment (ingar) [inline lib/doro/behaviors/god.ex]: Before this is supported in the HTML front end, this needs to be augmented with a `type`, or something like that, right?  Just wondering. 

```json
{
  "text": "You concentrate on the entity, etc.",
  "data": {
    "type": "open_entity_editor",
    "entity": {"id": "turntable" }
  }
}
```

```json
{
  "text": "You hear a sound.",
  "data": {
    "type": "play_sound",
    "url": "http://environmentalsounds.com/creepy_forest.wav"
  }
}
```

**Comment by bunnymatic** [comment]
deployed!
screen shot 2018-07-28 at 3 47 00 pm

**Pull request merged**